tokio-util/src/sync/cancellation_token.rs

```diff
@@ -118,15 +118,22 @@ impl Drop for CancellationToken {
         // next call
         let parent = inner.parent;
 
-        // Drop our own refcount
-        current_state = inner.decrement_refcount(current_state);
+        unsafe {
+            // Drop our own refcount
+            current_state = CancellationTokenState::decrement_refcount(inner, current_state);
```

Comment:
Here, `inner` is a mutable reference, but `decrement_refcount` might deallocate the memory. This would result in there being a dangling mutable reference, which is UB. More generally, you shouldn't call free on a raw pointer that's derived from a mutable reference - it's better to free the original raw pointer.

Reply:
Interesting. I agree, but Miri doesn't flag this. I wouldn't be surprised if the test case that should catch this is hidden behind an `epoll_create1` call.